src/eccs/Ecc.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of JKQ QFR library which is released under the MIT license.
+ * See file README.md or go to http://iic.jku.at/eda/research/quantum/ for more information.
+ */
+
+#include "eccs/Ecc.hpp"
+
+//#include <stdlib.h>
+
+Ecc::Ecc(struct EccInfo eccInfo, qc::QuantumComputation& quantumcomputation): ecc(eccInfo), qc(quantumcomputation) {
+}
+
+qc::QuantumComputation& Ecc::applyEcc() {
+    qc.stripIdleQubits(true, false);
+    statistics.nInputQubits = qc.getNqubits();
+	statistics.nOutputQubits = qc.getNqubits()*ecc.nRedundantQubits;
+	statistics.nOutputClassicalBits = qc.getNqubits()*ecc.nClassicalBits;
```

Comment:
Yes and no - it is currently wrong if the input circuit contains classical bits. Otherwise, for every ECC, measurements are taken in the end to correct each qubit. `ecc.nClassicalBits` denotes the number of classical bits needed by measurement per input circuit qubit. The total number of classical bits is therefore a multiple. 
However, 1) your question shows me that I should change the naming of the variable to make this more precise, and 2) I have to take care of input circuit classical bits - thanks!

Reply:
Thanks for clearing that up! Changing the name might already clear up most of this.
Typically, if the input circuit has `n` qubits, it will also contain `n` classical bits. I am not that familiar with the details of ECCs but I guess you can just extend these to `n*ecc.ClassicalBits` for the mapped circuit.